scio-test/src/main/scala/com/spotify/scio/testing/TransformOverride.scala

```diff
@@ -105,6 +106,38 @@ object TransformOverride {
     PTransformOverride.of(new EqualNamePTransformMatcher(name), overrideFactory)
   }
 
+  /**
+   * @return
+   *   A [[PTransformOverride]] which when applied will override a [[PTransform]] with name `name`
+   *   with a transform flat-mapping elements via `fn`.
+   */
+  def ofIter[T: ClassTag, U](name: String, fn: T => Iterable[U]): PTransformOverride = {
+    val wrappedFn: T => lang.Iterable[U] = fn
+      .compose { t: T =>
+        typeValidation(
+          s"Input for override transform $name does not match pipeline transform.",
+          t.getClass,
+          implicitly[ClassTag[T]].runtimeClass
+        )
+        t
+      }
+      .andThen(_.asJava)
+
+    val overrideFactory =
+      factory[PCollection[T], PCollection[U], PTransform[PCollection[T], PCollection[U]]](
+        t => PTransformReplacements.getSingletonMainInput(t),
+        new PTransform[PCollection[T], PCollection[U]]() {
+          override def expand(input: PCollection[T]): PCollection[U] = {
+            val inferableFn = new InferableFunction[T, lang.Iterable[U]] {
```

Comment:
I think `FlatMapElements#via` also accepts a `SimpleFunction` - you could leverage our utility class [Functions#simpleFn](https://github.com/spotify/scio/blob/main/scio-core/src/main/scala/com/spotify/scio/util/Functions.scala#L254) to wrap `wrappedFn` into the `SimpleFunction` Java interface and pass that to `via`.

Reply:
Didn't work for some reason, that's what I tried first. Please look at this: https://github.com/spotify/scio/pull/4484#discussion_r935819671